File: 2001/4/8-4.py
#!/usr/bin/env python

import sys

points = {0:500, 1:300, 2:250, 3:200, 4:150, 5:100, 6:75, 7:50, 8:25, 9:10, 10:0}

while True:
    line = sys.stdin.readline()
    if line == "":
        break
    name = line.split()[0]
    print
    print "Score Summary for", name
    print "-----------------------"
    score = 0
    for i in xrange(1, 6):
        line = sys.stdin.readline()
        line = line.split()
        x = float(line[0])
        y = float(line[1])
        dist = (x * x + y * y) ** 0.5
        d = int(dist / 50)
        if (dist % 50) == 0 and d > 0:
            d -= 1
        if d > 10:
            d = 10
        n = points[d]
        score += n
        print "   Hit %s =%13d" % (i, n)
    print "              ---------"
    print "   Score =%13d" % score

File: 2001/4/8-4.cpp
#include <iostream>
#include <fstream>
#include <iomanip>
#include <cmath>
#include <string>
#include <map>
using namespace std;

int main()
{
	ifstream in("prob4_in.txt");
	string name;
	map<int,int> points;
	points[0] = 500;
	points[1] = 300;
	points[2] = 250;
	points[3] = 200;
	points[4] = 150;
	points[5] = 100;
	points[6] = 75;
	points[7] = 50;
	points[8] = 25;
	points[9] = 10;
	while (in >> name)
	{
		cout << endl << "Score Summary for " << name << endl
			<< "-----------------------" << endl;
		int score = 0, d = 0, x = 0, y = 0, n = 0;
		double dist = 0;
		for (int i = 1; i < 6; i++)
		{
			in >> x >> y;
			dist = sqrt(x * x + y * y);
			d = dist / 50;
			if ((int)dist == dist)
				if ((((int)dist % 50) == 0) && (d > 0))
					d--;
			n = points[d];
			score += n;
			cout << "   Hit " << i << " =" << setw(13) << n << endl;
			d = 0;
		}
		cout << "              ---------" << endl
			<< "   Score =" << setw(13) << score << endl;
	}
}
